fastcrypto-zkp/src/api.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use std::io::Cursor;
+
+use ark_bls12_381::Bls12_381;
+use ark_groth16::{Proof, VerifyingKey};
+use ark_relations::r1cs::SynthesisError;
+use ark_serialize::CanonicalDeserialize;
+
+use crate::conversions::BlsFr;
+use crate::verifier::{process_vk_special, verify_with_processed_vk};
+
+/// Verify Groth16 proof using the serialized representation of verifying key, input and proof.
+/// This assumes canoncical ser/de defined in Arkworks, instead of the Zcash bytes representation.
+pub fn verify_groth16_in_bytes(
+    vk_bytes: Vec<u8>,
+    x_bytes: Vec<u8>,
+    proof_bytes: Vec<u8>,
+) -> Result<bool, SynthesisError> {
+    let reader_vk = Cursor::new(vk_bytes);
+    let reader_x = Cursor::new(x_bytes);
+    let reader_proof = Cursor::new(proof_bytes);
```

Comment:
It would be really cool if we could use `Untrusted` instead, which should allow panic-free APIs:
https://github.com/briansmith/untrusted
https://briansmith.org/rustdoc/untrusted/index.html